src/screens/JobProposal/SpecsView.test.tsx

```diff
@@ -136,16 +149,97 @@ describe('SpecsView', () => {
 
   describe('rejected proposal', () => {
     let specs: ReadonlyArray<JobProposal_SpecsFields>
+    let proposal: JobProposalPayloadFields
 
     beforeEach(() => {
+      proposal = buildJobProposal({ status: 'REJECTED' })
       specs = [buildJobProposalSpec({ status: 'REJECTED' })]
     })
 
     it('renders a rejected job proposal', async () => {
-      renderComponent(specs)
+      renderComponent(specs, proposal)
 
       expect(getByTestId('codeblock')).toHaveTextContent(specs[0].definition)
       expect(queryByText(/edit/i)).toBeNull()
     })
   })
+
+  describe('deleted proposal with approved spec', () => {
+    let specs: ReadonlyArray<JobProposal_SpecsFields>
+    let proposal: JobProposalPayloadFields
+
+    beforeEach(() => {
+      proposal = buildJobProposal({ status: 'DELETED' })
+      specs = [buildJobProposalSpec({ status: 'APPROVED' })]
+    })
+
+    it('renders a deleted job proposal', async () => {
+      renderComponent(specs, proposal)
+
+      expect(getByTestId('codeblock')).toHaveTextContent(specs[0].definition)
+      expect(queryByText(/edit/i)).toBeNull()
+    })
+
+    it('cancels the proposal', async () => {
+      renderComponent(specs, proposal)
+
+      userEvent.click(getByRole('button', { name: /cancel/i }))
+      userEvent.click(getByRole('button', { name: /confirm/i }))
+      await waitFor(() => expect(handleCancel).toHaveBeenCalled())
+    })
+  })
+
+  describe('deleted proposal with cancelled spec', () => {
+    let specs: ReadonlyArray<JobProposal_SpecsFields>
+    let proposal: JobProposalPayloadFields
+
+    beforeEach(() => {
+      proposal = buildJobProposal({ status: 'DELETED' })
+      specs = [buildJobProposalSpec({ status: 'CANCELLED' })]
+    })
+
+    it('renders a deleted job proposal', async () => {
```

Comment:
I see you use this pattern a couple times, can you add a docstring explaining what it's testing?  I'm not putting together in my mind the rendering with `Cancel` not being in the document.

Reply:
The render here is for the deleted proposals tab, which, after clicking, should display a list of the existing deleted proposals along with their specs once you click into a proposal (proposals can have many specs. When a proposal is deleted the specs don't get deleted but maintain whatever status they are, but can be changed based on user actions). Clicking on the cancel button on a spec will cancel the spec and also delete any running jobs. Only approved specs can be cancelled; otherwise, there's no additional actions to be done on a spec for a deleted proposal.